### PR TITLE
Docs finetuning callback example

### LINKED ISSUE #19711: How to use BackboneFinetuning callback?
### 📚 Documentation

Hi lightning team,

I am trying to implement finetuning of a custom model I have pretrained weights of.
It's a transformer architecture with a classification head on it.

I found this callback: https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.BackboneFinetuning.html#lightning.pytorch.callbacks.BackboneFinetuning

But i think this doc is not self sufficient as I don't really know what's expected from my model.
Do I need to have specifically named self.backbone and self.head attributes in my model class for it to work?

A more in-depth example could prove valuable here.

Thanks a lot in advance.

Antoine

cc @borda

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #19711
Adds documentation on how to use the `BackboneFinetuning` callback as it seems to confuse users how to format their model to work with the callback.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21216.org.readthedocs.build/en/21216/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by deependujha — APPROVED** [review]

**Comment by deependujha** [comment]
there seems to be doctest failure.
